From 5825d9d0bdb5c52481636fa4d31a54e055dc1856 Mon Sep 17 00:00:00 2001
From: Shengjiu Wang <shengjiu.wang@nxp.com>
Date: Fri, 2 Nov 2018 17:30:52 +0800
Subject: [PATCH 5009/5242] =?UTF-8?q?MLK-20183-1:=20ARM64:=20dts=EF=BC=9Arem?=
 =?UTF-8?q?ove=20the=20workaround=20for=20HDMI=20on=20imx8qm=20A0?=
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

commit  8e510f8f4cdd5662de845cda8d40254c396ad8b4 from
https://source.codeaurora.org/external/imx/linux-imx.git

Revert "MLK-17344-2: ARM64: dts: add constraint-rate for hdmi of imx8qm"

This reverts commit 86dbbb61cf05232e4c45bbd15759abd911dce28d.
On imx8qm B0 fix the DPLL jitter issue for HDMI module, so the constraint
for sample rate should be removed

Signed-off-by: Shengjiu Wang <shengjiu.wang@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 .../boot/dts/freescale/fsl-imx8qm-mek-hdmi.dts     |   23 +++++---------------
 1 file changed, 5 insertions(+), 18 deletions(-)

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek-hdmi.dts b/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek-hdmi.dts
index 711b35f..74e5aad 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek-hdmi.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek-hdmi.dts
@@ -22,7 +22,6 @@
 		compatible = "fsl,imx-audio-cdnhdmi";
 		model = "imx-audio-hdmi-tx";
 		audio-cpu = <&sai_hdmi_tx>;
-		constraint-rate = <48000>;
 		protocol = <1>;
 		hdmi-out;
 	};
@@ -68,26 +67,14 @@
 	status = "okay";
 };
 
-&amix {
-	status = "disabled";
-};
-
-&sai6 {
-	status = "disabled";
-};
-
-&sai7 {
-	status = "disabled";
-};
-
 &sai_hdmi_tx {
 	assigned-clocks =<&clk IMX8QM_ACM_HDMI_TX_SAI0_MCLK_SEL>,
-			<&clk IMX8QM_AUD_PLL1_DIV>,
-			<&clk IMX8QM_AUD_ACM_AUD_PLL_CLK1_DIV>,
-			<&clk IMX8QM_AUD_ACM_AUD_REC_CLK1_DIV>,
+			<&clk IMX8QM_AUD_PLL0_DIV>,
+			<&clk IMX8QM_AUD_ACM_AUD_PLL_CLK0_DIV>,
+			<&clk IMX8QM_AUD_ACM_AUD_REC_CLK0_DIV>,
 			<&clk IMX8QM_AUD_SAI_HDMITX0_MCLK>;
-	assigned-clock-parents = <&clk IMX8QM_AUD_ACM_AUD_PLL_CLK1_CLK>;
-	assigned-clock-rates = <0>, <768000000>, <768000000>, <768000000>, <768000000>;
+	assigned-clock-parents = <&clk IMX8QM_AUD_ACM_AUD_PLL_CLK0_CLK>;
+	assigned-clock-rates = <0>, <786432000>, <49152000>, <12288000>, <49152000>;
 	fsl,sai-asynchronous;
 	status = "okay";
 };
-- 
1.7.9.5

